dev/ProgressRing/ProgressRingLoading.cpp

```diff
@@ -0,0 +1,362 @@
+        result.StrokeEndCap(winrt::CompositionStrokeCap::Round);
+        result.StrokeStartCap(winrt::CompositionStrokeCap::Round);
+        result.IsStrokeNonScaling(true);
+        result.StrokeThickness(static_cast<float>(_strokeThickness));
+        return result;
+    }
+
+    // Layer (Shape): Radial
+    // Transforms: Radial
+    // Ellipse Path
+    // Ellipse Path.EllipseGeometry
+    // TrimStart
+    winrt::StepEasingFunction StepThenHoldEasingFunction()
+    {
+        auto result = _c.CreateStepEasingFunction();
+        result.IsInitialStepSingleFrame(true);
+        return result;
+    }
+
+    // Layer (Shape): Radial
+    // Transforms: Radial
+    // Ellipse Path
+    // Ellipse Path.EllipseGeometry
+    // TrimEnd
+    winrt::ScalarKeyFrameAnimation TrimEndScalarAnimation_0_to_0p5()
+    {
+        auto result = _c.CreateScalarKeyFrameAnimation();
+        result.Duration({ c_durationTicks });
+        result.InsertKeyFrame(0, 9.99999975E-05F, _holdThenStepEasingFunction);
+        result.InsertKeyFrame(0.5F, 0.5F, _cubicBezierEasingFunction);
+        return result;
+    }
+
+    // Layer (Shape): Radial
+    // Transforms: Radial
+    // Ellipse Path
+    // Ellipse Path.EllipseGeometry
+    // TrimStart
+    winrt::ScalarKeyFrameAnimation TrimStartScalarAnimation_0_to_0p5()
+    {
+        auto result = _c.CreateScalarKeyFrameAnimation();
+        result.Duration({ c_durationTicks });
+        result.InsertKeyFrame(0, 0, StepThenHoldEasingFunction());
+        result.InsertKeyFrame(0.5F, 0, HoldThenStepEasingFunction());
+        result.InsertKeyFrame(1, 0.5F, CubicBezierEasingFunction());
+        return result;
+    }
+
+public:
+    ProgressRingAnimatedVisual(winrt::Compositor compositor, double strokeThickness, winrt::Size size, winrt::Color foreground, winrt::Color background)
+        : _c(compositor),
+        _reusableExpressionAnimation(compositor.CreateExpressionAnimation()),
+        _strokeThickness(strokeThickness),
+        _size(size),
+        _foreground(foreground),
+        _background(background)
+    {
+        const auto _ = Root();
+    }
+
+    [[nodiscard]] auto RootVisual() const
+    {
+        return _root;
+    }
+
+    [[nodiscard]] winrt::float2 Size() const
+    {
+        return _size;
+    }
+
+    [[nodiscard]] std::chrono::milliseconds Duration() const
+    {
+        return { c_durationTicks };
+    }
+
+    void Close()
+    {
+        if (_root)
+        {
+            _root.Close();
+        }
+    }
+};
+
+ProgressRingLoading::ProgressRingLoading(double strokeThickness, winrt::Size size, winrt::Color foreground, winrt::Color background)
+    : m_strokeThickness(strokeThickness),
+    m_size(size),
+    m_foreground(foreground),
+    m_background(background)
+{
+
+}
+
+winrt::IAnimatedVisual ProgressRingLoading::TryCreateAnimatedVisual(
+    const winrt::Compositor& compositor,
+    winrt::IInspectable& diagnostics)
+{
+    diagnostics = nullptr;
+    if (!IsRuntimeCompatible())
```

Comment:
>IsRuntimeCom [](start = 9, length = 12)

Is it acceptable for the ring to not show up if you're running on a version of the OS that doesn't support Lottie? That's what this line will do.
What is the lowest OS version that you need to support?

Reply:
I believe the design team is still finalizing the designs, including the fallback. 
@YuliKl can most likely answer this question better than me.